pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/scheduler/SplitScheduler.java

```diff
@@ -0,0 +1,172 @@
+        this.context = context;
+        this.conf = pulsar.getConfiguration();
+        this.bundleSplitStrategy = bundleSplitStrategy;
+        this.serviceUnitStateChannel = serviceUnitStateChannel;
+    }
+
+    public SplitScheduler(PulsarService pulsar,
+                          ServiceUnitStateChannel serviceUnitStateChannel,
+                          SplitCounter counter,
+                          AtomicReference<List<Metrics>> splitMetrics,
+                          LoadManagerContext context) {
+        this(pulsar, serviceUnitStateChannel, counter, splitMetrics, context,
+                new DefaultNamespaceBundleSplitStrategyImpl(counter));
+    }
+
+    @Override
+    public void execute() {
+        boolean debugMode = conf.isLoadBalancerDebugModeEnabled() || log.isDebugEnabled();
+        if (debugMode) {
+            log.info("Load balancer enabled: {}, Split enabled: {}.",
+                    conf.isLoadBalancerEnabled(), conf.isLoadBalancerAutoBundleSplitEnabled());
+        }
+
+        if (!isLoadBalancerAutoBundleSplitEnabled()) {
+            if (debugMode) {
+                log.info("The load balancer or load balancer split already disabled. Skipping.");
+            }
+            return;
+        }
+
+        synchronized (bundleSplitStrategy) {
+            final Set<SplitDecision> decisions = bundleSplitStrategy.findBundlesToSplit(context, pulsar);
+            if (!decisions.isEmpty()) {
+                List<CompletableFuture<Void>> futures = new ArrayList<>();
+                for (SplitDecision decision : decisions) {
+                    if (decision.getLabel() == Success) {
+                        var split = decision.getSplit();
+                        futures.add(serviceUnitStateChannel.publishSplitEventAsync(split)
+                                .whenComplete((__, e) -> {
+                                    if (e == null) {
+                                        counter.update(decision);
+                                        log.info("Published Split Event for {}", split);
+                                    } else {
+                                        counter.update(Failure, Unknown);
+                                        log.error("Failed to publish Split Event for {}", split);
+                                    }
+                                }));
+                    }
+                }
+                FutureUtil.waitForAll(futures).exceptionally(ex -> {
+                    log.error("Failed to wait for split events to persist.", ex);
+                    counter.update(Failure, Unknown);
+                    return null;
+                });
+            } else {
+                if (debugMode) {
+                    log.info("BundleSplitStrategy returned no bundles to split.");
+                }
+            }
+        }
+
+        if (counter.updatedAt() > counterLastUpdatedAt) {
+            splitMetrics.set(counter.toMetrics(pulsar.getAdvertisedAddress()));
```

Comment:
Why don't we use `brokerRegistry().getBrokerId()`?

Reply:
To be consistent with other metrics code, here we use `pulsar.getAdvertisedAddress()`

Reply:
Ok, I see.